scripts/generate_verify_psa_macros.py

```diff
@@ -0,0 +1,168 @@
+        (up to non-significant whitespace) and skips all non-matching lines.
+
+        !!! Difference from generate_psa_constants.py !!!
+        This function has to handle line continuations left intact by read_file
+        """
+        m = re.match(self._define_directive_re, line)
+        if not m:
+            return
+        name, parameters, expansion = m.groups()
+        if parameters is None:
+            parameters = ''
+        expansion = re.sub(self._comment_re, ' ', expansion)
+        expansion = re.sub(self._trailing_whitespace_re, '', expansion)
+        macro = (parameters, expansion)
+        if expansion:
+            self.macros[name] = macro
+        else:
+            self.macros.setdefault(name, macro)
+
+    _nonascii_cr_re = re.compile(rb'[^\x00-\x7f\r]+')
+    _continued_line_re = re.compile(rb'\\\r?\n\Z')
+    def read_file(self, header_file):
+        """Scan header file for macro definitions
+
+        !!! Difference from generate_psa_constants.py !!!
+        This function leaves the line continuations intact
+        """
+        for line in header_file:
+            m = re.search(self._continued_line_re, line)
+            while m:
+                cont = next(header_file)
+                line = line[:m.end(0)] + cont
+                m = re.search(self._continued_line_re, line)
+            line = re.sub(self._nonascii_cr_re, rb'', line).decode('ascii')
+            self.read_line(line)
+
+    def write_file(self, output_file):
+        """Generate output from the gathered macros"""
+        header = []
+        tests = []
+
+        for name, (parameters, expansion) in self.macros.items():
+            macro = {'name' : name, 'parameters' : parameters, 'expansion' : expansion}
+            if not expansion or '_IS_' in name or '_GET_' in name:
+                tests.append(TEST_DEFINED_TEMPLATE.format_map(macro))
+            else:
+                header.append(HEADER_TEMPLATE.format_map(macro))
+                tests.append(TEST_TEMPLATE.format_map(macro))
+
+        output_file.write(OUTPUT_TEMPLATE.format(header='\n'.join(header), tests='\n'.join(tests)))
+
+def generate_verify_psa_macros(header_file_names, output_file_name):
+    collector = MacroCollector()
+    for header_file_name in header_file_names:
+        with open(header_file_name, 'rb') as header_file:
+            collector.read_file(header_file)
+    temp_file_name = output_file_name + '.tmp'
+    with open(temp_file_name, 'w') as output_file:
+        collector.write_file(output_file)
+    os.replace(temp_file_name, output_file_name)
+
+if __name__ == '__main__':
+    if not os.path.isdir('programs') and os.path.isdir('../programs'):
+        os.chdir('..')
+    generate_verify_psa_macros(('scripts/data_files/psa_spec/crypto.h',
+                                'scripts/data_files/psa_spec/specdef_values.rst'),
+                               'programs/test/verify_psa_macros.c')
```

Comment:
Is it useful to have `verify_psa_macros.c` as a test program checked into the repository? It seems to me like this program has no value except to compile it, run it and see its output as part of a test, so it could be generated and run on the fly like `test_psa_constant_names.py` does.